command/ca/rekey.go

```diff
@@ -0,0 +1,340 @@
+package ca
+
+import (
+	"github.com/smallstep/cli/crypto/pemutil"
+	"go.step.sm/crypto/keyutil"
+	"io/ioutil"
+	"math/rand"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/pki"
+	"github.com/smallstep/cli/command"
+	"github.com/smallstep/cli/errs"
+	"github.com/smallstep/cli/flags"
+	"github.com/smallstep/cli/ui"
+	"github.com/urfave/cli"
+)
+
+func rekeyCertificateCommand() cli.Command {
+	return cli.Command{
+		Name:   "rekey",
+		Action: command.ActionFunc(rekeyCertificateAction),
+		Usage:  "rekey a valid certificate",
+		UsageText: `**step ca rekey** <crt-file> <key-file>
+[**--out-cert**=<file>] [**--out-key**=<file>] [**--private-key**=<file>]
+[**--ca-url**=<uri>] [**--root**=<file>] [**--password-file**=<file>]
+[**--expires-in**=<duration>] [**--force**] [**--exec**=<string>] [**--daemon**]
+[**--expires-in**=<duration>] [**--pid**=<int>] [**--pid-file**=<file>]
+[**--signal**=<int>] [**--exec**=<string>] [**--renew-period**=<duration>]`,
+		Description: `
+**step ca rekey** command rekeys the given certificate (with a request to the
+certificate authority) and writes the new certificate and private key 
+to disk - either overwriting <crt-file> <key-file> or using a new file when
```

Comment:
either overwriting <crt-file> and <key-file> or writing to new files when the ...